src/Step.cc

```diff
@@ -180,6 +181,9 @@ void Step::set_indentation_level(short level)
 
 void Step::set_label(const std::string& label)
 {
+    if (gul14::trim_sv(label) != label) {
```

Comment:
You use the suffix [`..._sv`](https://gul14.info/namespacegul14.html#a8d4755d57b125523fefbadf2b4f6773e) to have a better memory efficiency?

Reply:
Well, without `_sv` it has to create a new string object. Possibly that is so short it will be SSO-ed, so not strictly memory efficience. It is just work we can avoid.
The `_sv` version just returns pointers to the original data (string) elements.

The not-`_sv` version calls the `_sv` version and creates a new string object from the two pointers (by copying).

Reply:
From the source:

```c++
namespace gul14 {


std::string trim(string_view str, string_view ws_chars)
{
    return std::string(trim_sv(str, ws_chars));
}

```

Reply:
As I read this ... we could technically allow any whitespace in the label except blank ` `. The non-blank whitespaces will be escaped in the serdes process, so they work. Just the blanks are not escaped and thus do not survive the general input parameter trimming on deserialization.
But maybe that makes the rules too complicated for the end user?

Rule for this PR: Label may not start or end in whitespace
Possible rule: Label may not start or end with blank (other whitespaces allowed)

Reply:
Hmm, in retrospective the solution is still rather inefficient.

It would be enough to check if the first (and last) character is a blank.

Like
```diff
 void Step::set_label(const std::string& label)
 {
-    if (gul14::trim_sv(label) != label) {
+    if (gul14::starts_with(label, ' ')) {
```

Reply:
That was just an example. Honestly I believe any optimization on setting the label is premature optimization. How often will that be called in a normal taskomat run? Almost never ;)